src/main/java/com/uid2/admin/Main.java

```diff
@@ -162,6 +176,7 @@ public void run() {
             AuthMiddleware auth = new AuthMiddleware(adminUserProvider);
             WriteLock writeLock = new WriteLock();
             IKeyGenerator keyGenerator = new SecureKeyGenerator();
+            IKeypairGenerator keypairGenerator = new SecureKeypairGenerator(IdentityScope.UID2, Environment.Production);
```

Comment:
How should these parameters be resolved?

Reply:
@thomasm-ttd - the reason why we need to know if it's UID2/EUID scope and Prod/integ Env is to add these info into the private/public key pair for CSTG as you asked me a few weeks ago and i documented this as part of requirement for this ticket [here](https://atlassian.thetradedesk.com/confluence/display/UID2/UID2-1374+Proposal%3A+Data+Schema+and+Endpoint+Changes+for+CSTG+Support). Could you help answer how we go about determining these params in admin service?

Reply:
There is a branch waiting to be merged (tjm-UID2-1275-key-formatting) that changes the config parameters to "admin_key_prefix": "UID2-A-L-"
The UID2 is the identity scope, and the -L- is the environment: L = Local, P = prod, I = Integ

Reply:
I have updated this to use the prefix strategy of tjm-UID2-1275-key-formatting. I think the two branches should stay seperate, though. This branch has no dependency on 1275 and remains only additions.

Reply:
that's ok for now